packages/editor-ui/src/components/Node/NodeCreator/composables/useActions.ts

```diff
@@ -214,13 +211,23 @@ export const useActions = () => {
 			OPEN_AI_NODE_MESSAGE_ASSISTANT_TYPE,
 		];
 
-		const isChatTriggerMissing =
-			allNodes.find((node) =>
-				[MANUAL_CHAT_TRIGGER_NODE_TYPE, CHAT_TRIGGER_NODE_TYPE].includes(node.type),
-			) === undefined;
 		const isCompatibleNode = addedNodes.some((node) => COMPATIBLE_CHAT_NODES.includes(node.type));
 
-		return isCompatibleNode && isChatTriggerMissing;
+		if (!isCompatibleNode) return false;
+
+		const { allNodes, getNodeTypes } = useWorkflowsStore();
+		const { getByNameAndVersion } = getNodeTypes();
+
+		// We want to add a trigger if there are no triggers other than Manual Triggers
+		const shouldAddChatTrigger = allNodes.every((node) => {
```

Comment:
I am a bit worried about the performance of `every` for large workflows. Can we use `some` or something similar that would stop as soon as it finds the trigger that is not the manual trigger?

Reply:
I had the same concern, which is why I moved `isCompatibleNode` up before the `chatTrigger` logic. 👍 This way this will only be run on adding compatible nodes.

`every` will also stop as soon as it hits the first false, so as soon as we encounter a trigger other than `MANUAL_TRIGGER`. (Though I appreciate larger workflows will scale non-triggers more than triggers usually)

`getByNameAndVersion` ends up calling a computed property, so we should reuse the one call to the backend to get node types anyway if I understand the code correctly.

Reply:
Should I add a comment in the code acknowledging this?

Reply:
Sure, thanks for clarification